### PR TITLE
Add Discussions, Discussion and DiscussionComment

### PR DESCRIPTION
**Pull request by keegancsmith**

### CODE REVIEW

File: sourcegraph/sourcegraph.proto [hunk 1]
```diff
@@ -59,6 +59,58 @@ message ListResponse {
 	bool has_more = 1 [(gogoproto.moretags) = "url:\",omitempty\""];
 }
 
+// Discussion stores information about a discussion
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: Minor, inconsistent that it's missing a period at the end of this sentence here (most other places have a period at the end).

File: sourcegraph/sourcegraph.proto [hunk 2]
```diff
@@ -59,6 +59,58 @@ message ListResponse {
...
+	// contains a commit component, but we track Discussions across
+	// commits. So all lookups normalize out the commit component, but the
+	// commit component records the commit at the time the Discussion was
+	// created.
+	graph.DefKey def_key = 5 [(gogoproto.nullable) = false];
+
+	// Ratings contains a list of users who have "starred" the discussion
+	repeated UserSpec ratings = 6;
+
+	// Comments contains comments by users in the discussion
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: Here and the one above too.

File: sourcegraph/sourcegraph.proto [hunk 3]
```diff
@@ -547,6 +599,26 @@ service Changesets {
 	rpc ListEvents(ChangesetSpec) returns (ChangesetEventList);
 }
 
+// Discussions is a service for discussing units in a repository
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: Here also.

File: sourcegraph/sourcegraph.proto [hunk 4]
```diff
@@ -547,6 +599,26 @@ service Changesets {
...
+service Discussions {
+	// Create creates a new Discussion and returns it, populating its
+	// fields, such as ID and CreatedAt.
+	rpc Create(Discussion) returns (Discussion);
+
+	// Get returns the Discussion by RepoSpec and ID.
+	rpc Get(DiscussionSpec) returns (Discussion);
+
+	// List lists changesets for a DefKey
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: Here.

File: sourcegraph/sourcegraph.proto [hunk 5]
```diff
@@ -547,6 +599,26 @@ service Changesets {
...
+	rpc List(DiscussionListOp) returns (DiscussionList);
+
+	// CreateComment creates a new DiscussionComment and returns it,
+	// populating its fields, such as ID and CreatedAt.
+	rpc CreateComment(DiscussionCommentCreateOp) returns (DiscussionComment);
+
+	// UpdateRating either adds or removes a star by a User
```
Comment (shurcooL) [inline sourcegraph/sourcegraph.proto]: Here the two below.

**Comment by shurcooL** [comment]
This PR is so great that the only possible room for improvement I found were a few missing periods at the end of a few sentences. I did not spot any other issues.

**Comment by keegancsmith** [comment]
Squashed and merged into nodb with commit edb91bd824d8cfa65515c92b1162188eaf090c34

**Pull request closed without merging**